Fix use generic identifier

albertinisg opened the pull request:
This PR fixes #64 

Note that this change must be done on top of #68

Commit:
Fix address, geo:point, aggregateRating and occupancyLevels elements

Commit:
Adapt the elements used for filtering functions

Commit:
Use a generic identifier for restaurants, reviews and reservations

Commit:
Modify the Reservation data model matching the new requirements

Commit:
Modify the Restaurant data model matching the new requirements

Commit:
Modify the Review data model matching the new requirements

Commit:
Add queryString option for PATCH and POST actions

Commit:
Start using 'keyValues' by default when listing elements from Orion

jmcanterafonseca commented on server/utils.js:
```diff
@@ -752,14 +774,26 @@ function getTimeBetweenDates(from, to) {
+  if (date) {
+  return crypto.createHash('sha1').update(name + date).digest('hex');
+  } else {
+  return crypto.createHash('sha1').update(name).digest('hex');
```
nit: check indentation

albertinisg replied:
True, my bad

jmcanterafonseca commented on server/utils.js:
```diff
@@ -632,36 +636,51 @@ function getOrgReservations(franchise, listOfRestaurants, listOfReservations) {
+      headers
+    );
+  } else {
+    return authRequest(
```
instead of doing this, apply this logic to the authRequest method to make queryString an optional parameter at that level and avoid this check that would have to be repeated everytime you use the same code pattern

albertinisg replied:
Agree, the check is already at authRequest function, so I'll remove it from here

jmcanterafonseca commented on server/utils.js:
```diff
@@ -632,36 +636,51 @@ function getOrgReservations(franchise, listOfRestaurants, listOfReservations) {
     uri += '/' + encodeURIComponent(element);
   }
+
+  if (typeof normalized === 'undefined') {
```
nit: if (!normalized)

jmcanterafonseca commented on server/utils.js:
```diff
@@ -632,36 +636,51 @@ function getOrgReservations(franchise, listOfRestaurants, listOfReservations) {
-function getListByType(type, element, headers) {
+function getListByType(type, element, headers, normalized) {
   var uri = '/v2/entities';
+  var options = {'type': type,'limit': '1000'};
```
shouldn't 1000 be a constant and a number?

albertinisg replied:
Yes, changing it!

jmcanterafonseca commented on server/utils.js:
```diff
@@ -464,44 +465,47 @@ function dataToSchema(listOfElements) {
+    schemaObject.location = {
+      'type': 'geo:point',
+      'value': geoObject.latitude + ', ' +
+        geoObject.longitude
```
I think they can go in the same line

jmcanterafonseca commented on server/utils.js:
```diff
@@ -464,44 +465,47 @@ function dataToSchema(listOfElements) {
 }
 
 function fixAddress(schemaObject, geoObject) {
+  // The returned object will be POST/PATCH(ed),
```
various comments here:

the name of the function seems to be misleading fix??
on the other hand you can use options=keyValues in updates as well

albertinisg replied:
Not really misleading, but what it does is, replace the address with the one received from the geocoder. As the addresses are not really well formatted and using upperCase in the json from openEuskadi, I decided to made this kind of 'enrichment'. Regarding the `keyValues` option, this function is called to add Restaurants into Orion, so we need to add them using the `value` field as agreed.
Still, if you think it is not worth it, I can remove it and keep the address as they are

jmcanterafonseca replied:
then fix is not a proper name, wouldn't it be completeAddress, nonetheless, please add comments they are always helpful

jmcanterafonseca commented on server/utils.js:
```diff
@@ -577,7 +581,7 @@ function getOrgRestaurants(org, listOfElements) {
   return objectToArray(listOfElements).filter(
     function(element) {
-      return element.author.name === user;
+      return element.author === user;
```
Couldn't these filters being implemented by means of NGSI v2 queries?

albertinisg replied:
Exactly, this is something I have planned for the next refactorization (all the methods from `orion.js` file)

jmcanterafonseca commented:
overall looks considerably better, @albertinisg could you respond to comments and fix suggestions you consider and then ask for a new review?

thanks

Commit:
Minor bugfixes and lint issues from review process

albertinisg commented:
Changes regarding the review process done in 023e29da79807ddac37e056256bea015d4c99db8. @jmcanterafonseca could you have a look? Thanks!

jmcanterafonseca commented on server/auth/authrequest.js:
```diff
@@ -54,9 +54,12 @@ function performRequest(endpoint, method, data, fiwareHeaders) {
       resolveWithFullResponse: true,
       json: true // Automatically stringifies the body to JSON
     };
+    if (querystring) {
```
code is duplicated below

Commit:
Remove duplicated code from authRequest

Commit:
Rename fixAddress function and add small description

albertinisg commented:
@jmcanterafonseca, removed duplicated code in 3bea80e70605262c0b189adbf86e2067bc78619c and renamed function in 7e693041e46693085aabf6dfc85615273dfd26e3

jmcanterafonseca commented on server/feeders/restaurantfeeder.js:
```diff
@@ -132,43 +133,54 @@ var feedOrionRestaurants = function() {
 
       if (element in addressDictionary) {
 
         if (val !== 'undefined' && val !== '') {
```
if (val) suffices 

it checks for both conditions at the same time

albertinisg replied:
Done in ff19ca7a280ee02896dd99dbd986debf336f3426

jmcanterafonseca commented on server/feeders/restaurantfeeder.js:
```diff
@@ -38,12 +38,12 @@ var fiwareHeaders = {
-  if (restaurant.address.addressLocality) {
-    address += restaurant.address.addressLocality + ' ';
+  var address = restaurant.address.value.streetAddress + ' ';
+  if (restaurant.address.value.addressLocality) {
```
I'm wondering why this trailing space is needed here??

albertinisg replied:
To format the address used in the geocoder

jmcanterafonseca commented on server/utils.js:
```diff
@@ -463,53 +464,60 @@ function dataToSchema(listOfElements) {
 
-function fixAddress(schemaObject, geoObject) {
+function completeAddress(schemaObject, geoObject) {
+  // The returned object will be POST/PATCH(ed),
```
nit: add the comment outside the function body and using /\* */

albertinisg replied:
Done in 24ebf0d3d65bd86f16690b17f4889dc6a932930b

jmcanterafonseca commented on server/utils.js:
```diff
@@ -752,14 +758,26 @@ function getTimeBetweenDates(from, to) {
 }
 
+function generateId(name, date) {
+  if (date) {
```
better if you create an intermediate variable, then you will save writing and make it clearer

albertinisg replied:
Done in 62a7ea877b8b08ea9cc3b045d99e6602fdbf0f17

Commit:
Fix checks

Commit:
Add comment outside the function

Commit:
Add intermediate variables

jmcanterafonseca commented:
LGTM
